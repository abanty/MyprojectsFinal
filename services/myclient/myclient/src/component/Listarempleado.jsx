import React from 'react';
 
const styles = {
    color: "black",
    marginTop: "50px",
    marginLeft: "0px",
    float: "center",
    fontSize: "18px"
};
const Listarempleado = (props) => {
  return (

    <div className="well" style={styles}> 
      <legend className="text-center header" >Lista de desaprobados</legend>
              <div>
                      <table className="table table-striped"  >
                         <thead className="thead-dark">
                            <tr>
                              <th scope="col">Usuario</th>
                              <th scope="col">Email</th> 
															<th scope="col">Direccion</th>                           
															<th scope="col">Telefono</th> 
															<th scope="col">Nota</th> 
                            </tr>
                          </thead>
													<tbody >
      {              
        props.users.map((user) => {
          return (                   
                    <tr key={user.id}>
                        <td>{user.username}</td>
                        <td>{user.email}</td>
												<td>{user.address}</td>  
												<td>{user.phone}</td>  
												<td>{user.age}</td>                        
                    </tr>                    
          )
        })
			}
							</tbody> 
                </table>                     
        </div>
    </div> 
  )
};
 
export default Listarempleado;