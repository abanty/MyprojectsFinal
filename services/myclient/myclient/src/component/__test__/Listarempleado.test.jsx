import React from 'react';
import { shallow } from 'enzyme'; 
import Listarempleado from '../Listarempleado';
import renderer from 'react-test-renderer';
 
const users = [
  {
	'active': true,
	'email': 'aprobadocontrece@gmail.com',
	'id': 1,
  'username': 'abanto',
  'address': 'alameda',
  'phone': '991644787',
  'age': '25'
  },
  {
	'active': true,
	'email': 'aprobadoconquince@upeu.edu.pe',
	'id': 2,
  'username': 'omar',
  'address': 'inti',
  'phone': '991111187',
  'age': '21'
  }
];
 
test('Listarempleado renders properly', () => {
  const wrapper = shallow(<Listarempleado users={users}/>);
  const element = wrapper.find('th');
  expect(element.length).toBe(5);
  expect(element.get(0).props.children).toBe('Usuario');
});
 
test('Listarempleado snapshot properly', () => {
  const tree = renderer.create(<Listarempleado users={users}/>).toJSON();
  expect(tree).toMatchSnapshot();  
});


