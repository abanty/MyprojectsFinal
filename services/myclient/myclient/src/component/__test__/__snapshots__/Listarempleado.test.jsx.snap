// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listarempleado snapshot properly 1`] = `
<div
  className="well"
  style={
    Object {
      "color": "black",
      "float": "center",
      "fontSize": "18px",
      "marginLeft": "0px",
      "marginTop": "50px",
    }
  }
>
  <legend
    className="text-center header"
  >
    Lista de desaprobados
  </legend>
  <div>
    <table
      className="table table-striped"
    >
      <thead
        className="thead-dark"
      >
        <tr>
          <th
            scope="col"
          >
            Usuario
          </th>
          <th
            scope="col"
          >
            Email
          </th>
          <th
            scope="col"
          >
            Direccion
          </th>
          <th
            scope="col"
          >
            Telefono
          </th>
          <th
            scope="col"
          >
            Nota
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            abanto
          </td>
          <td>
            aprobadocontrece@gmail.com
          </td>
          <td>
            alameda
          </td>
          <td>
            991644787
          </td>
          <td>
            25
          </td>
        </tr>
        <tr>
          <td>
            omar
          </td>
          <td>
            aprobadoconquince@upeu.edu.pe
          </td>
          <td>
            inti
          </td>
          <td>
            991111187
          </td>
          <td>
            21
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
